File: benchmarks/sieve/sieve.py
n = 1000000
prime = []

for i in range(n + 1):
    prime.append(True)

prime[0] = False
prime[1] = False

i = 2
while i * i <= n:
    if prime[i]:
        j = i * i
        while j <= n:
            prime[j] = False
            j += i
    i += 1

count = 0

for i in range(n + 1):
    if prime[i]:
        count += 1

print(count)

File: benchmarks/sieve/sieve.js
const n = 1000000
const prime = []

for (let i = 0; i <= n; i++) {
	prime.push(true)
}

prime[0] = false
prime[1] = false

for (let i = 2; i * i <= n; i++) {
	if (prime[i]) {
		for (let j = i * i; j <= n; j += i) {
			prime[j] = false
		}
	}
}

let count = 0

for (let i = 0; i <= n; i++) {
	if (prime[i]) count++
}

console.log(count)
